client/allocdir/alloc_dir.go

```diff
@@ -400,7 +402,13 @@ func fileCopy(src, dst string, perm os.FileMode) error {
 	defer dstFile.Close()
 
 	if _, err := io.Copy(dstFile, srcFile); err != nil {
-		return fmt.Errorf("Couldn't copy %v to %v: %v", src, dst, err)
+		return fmt.Errorf("Couldn't copy %q to %q: %v", src, dst, err)
+	}
+
+	if uid >= 0 && gid >= 0 {
```

Comment:
I want to make sure they're both both valid IDs. Even if it's `root:bob` that's uid=0, gid>0 which will pass this conditional.